spec/unit/file_system_spec.rb

```diff
@@ -999,6 +999,15 @@ def increment_counter_in_multiple_processes(file, num_procs, options)
             Puppet::FileSystem.replace_file(dest, 0755) { |_| }
           }.to raise_error(ArgumentError, /Only modes 0644, 0640, 0660, and 0440 are allowed/)
         end
+
+        it 'falls back to fully qualified user name when sid retrieval fails' do
+          current_user_sid = Puppet::Util::Windows::SID.name_to_sid(Puppet::Util::Windows::ADSI::User.current_user_name)
+          allow(Puppet::Util::Windows::SID).to receive(:name_to_sid).with(Puppet::Util::Windows::ADSI::User.current_user_name).and_return(nil, current_user_sid)
```

Comment:
I think that inside tests we should use `expect` instead `allow`

Reply:
I totally agree but this aligns with the rest of the tests in this file. On previous occasions, we agreed as a team that consistency is more important.